docs/index.rst

```diff
@@ -24,6 +24,7 @@ You may install JAXopt from GitHub with:
    stochastic
    implicit_diff
    root_finding
+   auto_examples/index
```

Comment:
Technically, examples are not part of the public API. So I would prefer

**Public API**

* Basics
* Unconstrained optimization
* Constrained optimization
* Non-smooth optimization
* Stochastic optimization
* Implicit differentiation
* Root finding

**Examples**

- ...
- ...
- ...

Reply:
![Screenshot-20210903175942-960x1056](https://user-images.githubusercontent.com/277639/132068968-eea4ceec-dc7a-4dbb-bb23-60e6a3b6aa22.png)

How about the above? Almost what you want, but without expanding the examples. I think we should do this once we start putting the examples into different folders (as sklearn does).

Reply:
Why is it necessary to put examples in different folders?

Reply:
Because I don't know how to expand the examples otherwise. I tried changing the :maxdepth: in the toctree, like 
```
.. toctree::
   :maxdepth: 3
   :caption: Examples

   auto_examples/index
```

but it doesn't seem to change anything

Reply:
I see! I'm fine with the current solution for now then.